### PR TITLE
Empty string DOI.doi in index.

### PR DESCRIPTION
**Pull request by fightingtexasaggie**
If there's a better way to explicitly set to null, please chime in. $document->set('somenestedindex', null) doesn't work. I'm not terribly happy about this hackish emptystring in a new DOI object, but it works.

### CODE REVIEW

File: src/Event/DatasetIndexSubscriber.php [hunk 1]
```diff
@@ -119,6 +120,12 @@ public function populateCalculatedFields(TransformEvent $event)
                 $document->set('collectionEndDate', $collectionEndDate->format('Y-m-d H:i:s'));
             }
         }
+
+        // For DIF only, set DOI to a DOI with empty string as the doi value.
+        if (false === $dataset->hasDatasetSubmission()) {
```
Comment (praneethpr) [inline src/Event/DatasetIndexSubscriber.php:125]: We usually use instanceOf to check if its an object and if its not null, and we also need to check if the dataset has been submitted

File: src/Event/DatasetIndexSubscriber.php [hunk 2]
```diff
@@ -119,6 +120,12 @@ public function populateCalculatedFields(TransformEvent $event)
...
+                $doi = new DOI('');
```
Comment (praneethpr) [inline src/Event/DatasetIndexSubscriber.php:126]: You don't need to do this, you can get the DOI from the dataset ($dataset->getDoi())

**Comment by mickel1138** [comment]
I'm not sure this is the right approach. This should probably be done in the front end?

File: src/Event/DatasetIndexSubscriber.php (hunk 1 above)
Comment (fightingtexasaggie) [reply src/Event/DatasetIndexSubscriber.php:125]: hasDatasetSubmission returns a bool, so used it over $dataset->getDatasetSubmission(). I forgot about the unsubmitted...will fix.

File: src/Event/DatasetIndexSubscriber.php (hunk 2 above)
Comment (fightingtexasaggie) [reply src/Event/DatasetIndexSubscriber.php]: indeed correct, changed.

**Comment by mickel1138** [comment]
> Doi doesn't show on any of them now

Yes, I verified this too. I didn't turn of the Identified filter.

**Comment by fightingtexasaggie** [comment]
Closing PR. We agreed fixing in the view is a better solution, instead of custom indexing.

**Pull request closed without merging**